hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/BasicOmKeyInfo.java

```diff
@@ -0,0 +1,189 @@
+public class BasicOmKeyInfo {
+
+  private String volumeName;
+  private String bucketName;
+  private String keyName;
+  private long dataSize;
+  private long creationTime;
+  private long modificationTime;
+  private ReplicationConfig replicationConfig;
+  private boolean isFile;
+
+  @SuppressWarnings("parameternumber")
+  public BasicOmKeyInfo(String volumeName, String bucketName, String keyName,
+                        long dataSize, long creationTime, long modificationTime,
+                        ReplicationConfig replicationConfig, boolean isFile) {
+    this.volumeName = volumeName;
+    this.bucketName = bucketName;
+    this.keyName = keyName;
+    this.dataSize = dataSize;
+    this.creationTime = creationTime;
+    this.modificationTime = modificationTime;
+    this.replicationConfig = replicationConfig;
+    this.isFile = isFile;
+  }
+
+  public String getVolumeName() {
+    return volumeName;
+  }
+
+  public String getBucketName() {
+    return bucketName;
+  }
+
+  public String getKeyName() {
+    return keyName;
+  }
+
+  public long getDataSize() {
+    return dataSize;
+  }
+
+  public long getCreationTime() {
+    return creationTime;
+  }
+
+  public long getModificationTime() {
+    return modificationTime;
+  }
+
+  public ReplicationConfig getReplicationConfig() {
+    return replicationConfig;
+  }
+
+  public boolean isFile() {
+    return isFile;
+  }
+
+  /**
+   * Builder of BasicOmKeyInfo.
+   */
+  public static class Builder {
+    private String volumeName;
+    private String bucketName;
+    private String keyName;
+    private long dataSize;
+    private long creationTime;
+    private long modificationTime;
+    private ReplicationConfig replicationConfig;
+    private boolean isFile;
+
+    public Builder setVolumeName(String volumeName) {
+      this.volumeName = volumeName;
+      return this;
+    }
+
+    public Builder setBucketName(String bucketName) {
+      this.bucketName = bucketName;
+      return this;
+    }
+
+    public Builder setKeyName(String keyName) {
+      this.keyName = keyName;
+      return this;
+    }
+
+    public Builder setDataSize(long dataSize) {
+      this.dataSize = dataSize;
+      return this;
+    }
+
+    public Builder setCreationTime(long creationTime) {
+      this.creationTime = creationTime;
+      return this;
+    }
+
+    public Builder setModificationTime(long modificationTime) {
+      this.modificationTime = modificationTime;
+      return this;
+    }
+
+    public Builder setReplicationConfig(ReplicationConfig replicationConfig) {
+      this.replicationConfig = replicationConfig;
+      return this;
+    }
+
+    public Builder setIsFile(boolean isFile) {
+      this.isFile = isFile;
+      return this;
+    }
+
+    public BasicOmKeyInfo build() {
+      return new BasicOmKeyInfo(volumeName, bucketName, keyName, dataSize,
+              creationTime, modificationTime, replicationConfig, isFile);
+    }
+  }
+
+  public BasicKeyInfo getProtobuf() {
```

Comment:
nit: not properly indented with 4 spaces for continuation of line.
```suggestion
  public BasicKeyInfo getProtobuf() {
    BasicKeyInfo.Builder builder = BasicKeyInfo.newBuilder()
        .setVolumeName(volumeName)
        .setBucketName(bucketName)
        .setKeyName(keyName)
        .setDataSize(dataSize)
        .setCreationTime(creationTime)
        .setModificationTime(modificationTime)
        .setType(replicationConfig.getReplicationType());
    if (replicationConfig instanceof ECReplicationConfig) {
      builder.setEcReplicationConfig(
          ((ECReplicationConfig) replicationConfig).toProto());
    } else {
      builder.setFactor(ReplicationConfig.getLegacyFactor(replicationConfig));
    }
    builder.setIsFile(isFile);

    return builder.build();
  }
```

Not sure if your IDE is configured with [ozone-coding-style](https://github.com/apache/ozone/blob/master/hadoop-ozone/dev-support/intellij/ozone-style.xml). If not, please add this to your IDE's code style.

Reply:
Thank you @hemantk-12, I missed this. I made the required changes.